mne/minimum_norm/inverse.py

```diff
@@ -1122,7 +1122,7 @@ def _apply_inverse_epochs_gen(epochs, inverse_operator, lambda2, method='dSPM',
                 sol *= noise_norm
         else:
             # Linear inverse: do computation here or delayed
-            if len(sel) < K.shape[1]:
+            if delayed:
```

Comment:
This logic branching does not seem to be complete. It seems like it should be:

- `if delayed and is_free_ori pick_ori != 'vector'`, raise an error
- All other conditions should be workable: return the "correct thing" (either a K/e[sel] tuple if delayed, else the full solution)

Reply:
But what about the previous part?
```python
        if is_free_ori:
            # Compute solution and combine current components (non-linear)
            sol = np.dot(K, e[sel])  # apply imaging kernel
            if pick_ori != 'vector':
                logger.info('combining the current components...')
                sol = combine_xyz(sol)
```
The `combine_xyz` after `np.dot` looks tricky to me...

Reply:
That's the code path that I'm saying should raise an error. The code path `is_free_ori and pick_ori == 'vector'` should work in a kernelized way. Currently it does not, and it also does not raise an error saying `delayed=True` will be ignored/not done.

Reply:
Oh! yes, sorry, i mistook the statement...

Reply:
@larsoner it works now. Thanks for the hint!